pkg/application-grid-controller/util/template_hash.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2020 The SuperEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"github.com/davecgh/go-spew/spew"
+	"github.com/superedge/superedge/pkg/application-grid-controller/controller/common"
+	"hash"
+	"hash/fnv"
+)
+
+func GenerateHash(template interface{}) uint64 {
```

Comment:
Uint64 has a small range and is prone to hash collision. Please pay attention to user feedback